### PR TITLE
[#2924] Refactor datasources

### LINKED ISSUE #2924: Add datasource active record
Documentation:
https://docs.cyfronet.pl/pages/viewpage.action?spaceKey=FID&title=Datasource+as+a+service

Add datasource active record which would inherit from service and would fulfill polymorphic requirements. 

Datasource has following additional fields compared to Service. They should be stored in the Datasource record.

```
"submissionPolicyURL": "https://example.com/",
      "preservationPolicyURL": "https://example.com/",
      "versionControl": false,
      "persistentIdentitySystems": [
        {
          "persistentIdentityEntityType": "ds_research_entity_type-organizations",
          "persistentIdentityEntityTypeSchemes": [
            "ds_persistent_identity_scheme-ark"
          ]
        },
        {
          "persistentIdentityEntityType": "ds_research_entity_type-organizations",
          "persistentIdentityEntityTypeSchemes": [
            "ds_persistent_identity_scheme-doi"
          ]
        }
      ],
      "jurisdiction": "ds_jurisdiction-global",
      "datasourceClassification": "ds_classification-repository",
      "researchEntityTypes": [
        "ds_research_entity_type-funders"
      ],
      "thematic": true,
      "researchProductLicensings": [
        {
          "researchProductLicenseName": "string",
          "researchProductLicenseURL": "https://upload.wikimedia.org/wikipedia/commons/7/77/Nederland_1_Logo_2000.png"
        },
        {
          "researchProductLicenseName": "ggggg",
          "researchProductLicenseURL": "https://google.com/"
        }
      ],
      "researchProductAccessPolicies": [
        "ds_coar_access_rights_1_0-embargoed_access",
        "ds_coar_access_rights_1_0-open_access"
      ],
      "researchProductMetadataLicensing": {
        "researchProductMetadataLicenseName": "in",
        "researchProductMetadataLicenseURL": "https://in.gr/"
      },
      "researchProductMetadataAccessPolicies": [
        "ds_coar_access_rights_1_0-metadata_only_access",
        "ds_coar_access_rights_1_0-restricted_access"
      ]
```

The scope of this issue is to add an active record and importer functionality which would be able to handle datasources.
All views should work "as is" without requirements of changing them.

- [ ] script update

### PR DESCRIPTION
**Pull request by goreck888**
Change datasource model as a Service subclass
unified schema for table services
migrated data from table datasources to services
Closes #2924

### CODE REVIEW

**Comment by Marcelinna** [comment]
1. Go to https://marketplace-6.docker-fid.grid.cyf-kr.edu.pl/backoffice/services
2. Click "Create new Service"
3. I see field for Datasources
image

**Comment by Marcelinna** [comment]
1. Go to https://marketplace-6.docker-fid.grid.cyf-kr.edu.pl/backoffice/services
2. Click "Create new Service"
3. Select datasources
4. Fill required field
5. Click on button "Create Service"

image

**Comment by Marcelinna** [comment]
- [ ] QA
- [ ] DEV
Problem with some fields, result aren't visible 
image

**Comment by Marcelinna** [comment]
- [ ] QA
- [ ] DEV

1. Go to https://marketplace-6.docker-fid.grid.cyf-kr.edu.pl/backoffice/services
2. Select data source
3. Click create service without fill the required fields - why Jurisdiction is required field, shouldn't be
image

**Comment by Marcelinna** [comment]
- [ ] QA
- [ ] DEV

1. Go to https://marketplace-6.docker-fid.grid.cyf-kr.edu.pl/backoffice/services
2. Select data source
3. Don't fill all required fields
4. Click Create service - validation doesn't allow save datasource
5. Fill all required filed 
6. Click Create service - form doesn't allow save data source some filed are empty, despite they are filled
image
image

**Comment by eldenkat** [comment]
Preview button doesn't work:

**Comment by eldenkat** [comment]
1. Access the service doesn't work. After creating a project  I've tried to pin the service (datasource), but failed: 



2. So, I've tried to add the same service to the same project (failed previously) and that's what I got:

**Comment by Marcelinna** [comment]
- [ ] QA 
- [ ] DEV
I change status pending to approved for datasources created in PC, but it is not visible in MP
image
image

**Comment by Marcelinna** [comment]
- [ ] QA
- [ ] DEV

label should be horizontal services
image

**Comment by Marcelinna** [comment]
- [ ] QA
- [ ] DEV
duplicate value
image

**Comment by Marcelinna** [comment]
- [ ] QA
- [ ] DEV
I can't import datasource
eosc.provider_1205.82544372c4d7f574944c451839c0b521

**Review by Michal-Kolomanski — APPROVED** [review]
Looks good to me. Well done :muscle:

**Comment by Michal-Kolomanski** [comment]
@goreck888 It was decided that in SS -  `all catalogs` and `services` tabs there will be `both services and data sources` together. 
However in the `data sources tab` - there will be `only data sources`. Therefore, please keep services and data sources separate in the script that provide data for SS.

File: app/services/importers/datasource.rb [hunk 1]
```diff
@@ -140,6 +140,7 @@ def call
     main_contact = MainContact.new(map_contact(@data["mainContact"])) if @data["mainContact"]
 
     {
+      type: "Datasource",
```
[the remark this answers is not in the document]
Comment (goreck888) [reply app/services/importers/datasource.rb]: Done.

File: app/services/importers/token.rb [hunk 2]
```diff
@@ -17,8 +17,7 @@ def initialize(
   end
 
   AAI_BASE_URL = "https://#{ENV["IMPORTER_AAI_BASE_URL"] || ENV["CHECKIN_HOST"] || "aai.eosc-portal.eu"}".freeze
-  AAI_TOKEN_PATH = "/oidc/token"
-  SCOPE = %w[openid profile email].freeze
+  AAI_TOKEN_PATH = "/auth/realms/core/protocol/openid-connect/token"
```
[the remark this answers is not in the document]
Comment (goreck888) [reply app/services/importers/token.rb:20]: It's hardcoded now but it can be fixed in the next step of development. Create a ticket for that.

**Pull request merged**